tap_postgres/sync_strategies/incremental.py

```diff
@@ -52,6 +63,12 @@ def sync_table(conn_info, stream, state, desired_columns, md_map):
     replication_key_value = singer.get_bookmark(state, stream['tap_stream_id'], 'replication_key_value')
     replication_key_sql_datatype = md_map.get(('properties', replication_key)).get('sql-datatype')
```

Comment:
I tried to run it and gives an error at this point:

```
replication_key_sql_datatype = md_map.get(('properties', replication_key)).get('sql-datatype')
AttributeError: 'NoneType' object has no attribute 'get
```

Seems like we need another:
```
if replication_key not in RESERVED_REPLICATION_KEYS:
    replication_key_sql_datatype = # Or define RESERVED_REPLICATION_KEYS as list of dicts with sql_datatype property 
else:
    replication_key_sql_datatype = md_map.get(('properties', replication_key)).get('sql-datatype')
```